Distutils removed in Python 3.13.

[Python 3.13 removed distutils](https://docs.python.org/3.12/library/distutils.html). This makes optys unittests fail under 3.13.

I get this error when running `pytest -vv opty/tests/test_direct_collocation.py` in the `opty-dev` environment on Windows:

```
================================================= test session starts =================================================
platform win32 -- Python 3.13.1, pytest-8.3.4, pluggy-1.5.0
rootdir: ..\opty
plugins: cov-6.0.0
collected 0 items / 1 error

======================================================= ERRORS ========================================================
_______________________________ ERROR collecting opty/tests/test_direct_collocation.py ________________________________
ImportError while importing test module '..\opty\opty\tests\test_direct_collocation.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
..\..\AppData\Local\anaconda3\envs\opty-dev\Lib\importlib\__init__.py:88: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
opty\__init__.py:4: in <module>
    from .direct_collocation import *
opty\direct_collocation.py:20: in <module>
    from .utils import (ufuncify_matrix, parse_free, _optional_plt_dep,
opty\utils.py:11: in <module>
    from distutils.ccompiler import new_compiler
E   ModuleNotFoundError: No module named 'distutils'
=============================================== short test summary info ===============================================
ERROR opty/tests/test_direct_collocation.py
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
================================================== 1 error in 1.90s ===================================================
```

The tests run under 3.12.

--- a/README.rst
+++ b/README.rst
@@ -79,6 +79,7 @@ The required dependencies are as follows:
 - numpy >= 1.19.0
 - python 3.9-3.13
 - scipy >= 1.5.0
+- setuptools
 - sympy >= 1.6.0
 
 The optional dependencies are as follows:
@@ -136,7 +137,7 @@ set the ``LD_LIBRARY_PATH`` so that you can link to Ipopt when installing
 
 Once Ipopt is installed and accessible, install conda then create an environment::
 
-   $ conda create -n opty-custom -c conda-forge cython numpy pip scipy sympy
+   $ conda create -n opty-custom -c conda-forge cython numpy pip scipy setuptools sympy
    $ source activate opty-custom
    (opty-custom)$ pip install cyipopt  # this will compile cyipopt against the available ipopt
    (opty-custom)$ pip install opty
@@ -145,7 +146,7 @@ If you want to develop opty, create a conda environment with all of the
 dependencies installed::
 
    $ conda config --add channels conda-forge
-   $ conda create -n opty-dev python sympy numpy scipy cython ipopt cyipopt matplotlib pytables pydy pandas pytest sphinx sphinx-gallery numpydoc
+   $ conda create -n opty-dev python sympy numpy scipy cython ipopt cyipopt matplotlib pytables pydy pandas pytest setuptools sphinx sphinx-gallery numpydoc
    $ source activate opty-dev
 
 Next download the opty source files and install with::

--- a/opty-dev-env.yml
+++ b/opty-dev-env.yml
@@ -13,6 +13,7 @@ dependencies:
   - pytest-cov
   - python
   - scipy >=1.5.0
+  - setuptools
   - sphinx
   - sphinx-gallery
   - sympy >=1.13  # version required for the docs

--- a/setup.py
+++ b/setup.py
@@ -20,6 +20,7 @@ setup(
         'cython>=0.29.19',
         'numpy>=1.19.0',
         'scipy>=1.5.0',
+        'setuptools',  # provides distutils for Python >=3.13
         'sympy>=1.6.0',
     ],
     extras_require={

